zerver/actions/streams.py

```diff
@@ -526,21 +527,34 @@ def bulk_add_subscriptions(
                 if sub.active:
                     already_subscribed.append(sub_info)
                 else:
+                    if not stream.push_notifications_enabled:
+                        sub_info.sub.push_notifications = None
+                    else:
+                        sub_info.sub.push_notifications = True
                     subs_to_activate.append(sub_info)
 
+        subs = [info.sub for info in subs_to_activate]
+        Subscription.objects.bulk_update(subs, ["push_notifications"])
```

Comment:
I think we should call `bulk_update` outside the `for user_profile in users:` loop so that we can just update all the `Subscription` objects by single update to database.